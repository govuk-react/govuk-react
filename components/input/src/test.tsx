import React from 'react';
import { mount } from 'enzyme';
import { ORANGE } from 'govuk-colours';

import Input from '.';

describe('Input', () => {
  const wrapper = <Input type="text" />;
  const withError = <Input type="text" error />;
  const withColouredError = <Input type="text" errorColor={ORANGE} error />;

  it('matches wrapper snapshot', () => {
    expect(mount(wrapper)).toMatchSnapshot('wrapper mount');
  });

  it('matches withError snapshot', () => {
    expect(mount(withError)).toMatchSnapshot('with error mount');
  });

  it('matches withColouredError snapshot', () => {
    expect(mount(withColouredError)).toMatchSnapshot('with coloured error mount');
  });
});
