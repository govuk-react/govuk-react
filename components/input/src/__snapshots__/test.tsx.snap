// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input matches withColouredError snapshot: with coloured error mount 1`] = `
.c0 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  box-sizing: border-box;
  width: 100%;
  height: 40px;
  margin-top: 0;
  padding: 5px;
  border: 2px solid #0b0c0c;
  border-radius: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 4px solid #f47738;
  margin-bottom: 0;
}

.c0:focus {
  outline: 3px solid #ffdd00;
  outline-offset: 0;
}

.c0::-webkit-outer-spin-button,
.c0::-webkit-inner-spin-button {
  margin: 0;
  -webkit-appearance: none;
}

.c0[type="number"] {
  -moz-appearance: textfield;
}

@media print {
  .c0 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-bottom: 0;
  }
}

<Input
  error={true}
  errorColor="#f47738"
  type="text"
>
  <input
    className="c0"
    type="text"
  />
</Input>
`;

exports[`Input matches withError snapshot: with error mount 1`] = `
.c0 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  box-sizing: border-box;
  width: 100%;
  height: 40px;
  margin-top: 0;
  padding: 5px;
  border: 2px solid #0b0c0c;
  border-radius: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 4px solid #d4351c;
  margin-bottom: 0;
}

.c0:focus {
  outline: 3px solid #ffdd00;
  outline-offset: 0;
}

.c0::-webkit-outer-spin-button,
.c0::-webkit-inner-spin-button {
  margin: 0;
  -webkit-appearance: none;
}

.c0[type="number"] {
  -moz-appearance: textfield;
}

@media print {
  .c0 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-bottom: 0;
  }
}

<Input
  error={true}
  type="text"
>
  <input
    className="c0"
    type="text"
  />
</Input>
`;

exports[`Input matches wrapper snapshot: wrapper mount 1`] = `
.c0 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  box-sizing: border-box;
  width: 100%;
  height: 40px;
  margin-top: 0;
  padding: 5px;
  border: 2px solid #0b0c0c;
  border-radius: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  margin-bottom: 0;
}

.c0:focus {
  outline: 3px solid #ffdd00;
  outline-offset: 0;
}

.c0::-webkit-outer-spin-button,
.c0::-webkit-inner-spin-button {
  margin: 0;
  -webkit-appearance: none;
}

.c0[type="number"] {
  -moz-appearance: textfield;
}

@media print {
  .c0 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-bottom: 0;
  }
}

<Input
  error={false}
  type="text"
>
  <input
    className="c0"
    type="text"
  />
</Input>
`;
